js/src/views/Application/TabBar/tabBar.js

```diff
@@ -140,12 +190,115 @@ export default class TabBar extends Component {
         value={ isTest ? 'TEST' : netChain } />
       );
 
-    return this.renderLabel('status', bubble);
+    return this.renderLabel(label, bubble);
   }
 
   onActivate = (tab) => {
     const { router } = this.context;
+    const activeRoute = tab.props['data-route'];
+
+    router.push(activeRoute);
+    this.setState(activeRoute);
+  }
+
+  toggleMenu = (event, menu) => {
+    const id = menu.props['data-id'];
+    const toggle = `${id}Visible`;
+    const isActive = this.state[toggle];
+
+    if (this.tabs[id].fixed) {
+      return;
+    }
+
+    this.setState({
+      [toggle]: !isActive
+    }, this.saveViews);
+  }
+
+  getDefaultViews () {
+    const views = {};
+
+    Object.keys(this.tabs).forEach((id) => {
+      const tab = this.tabs[id];
+
+      views[id] = {
+        active: tab.active || false
+      };
+    });
 
-    router.push(tab.props['data-route']);
+    return views;
+  }
+
+  loadViews () {
+    const defaults = this.getDefaultViews();
+    const state = {};
+    let lsdata;
+
+    try {
+      const json = window.localStorage.getItem(LS_VIEWS) || {};
+
+      lsdata = Object.assign(defaults, JSON.parse(json));
+    } catch (e) {
+      lsdata = defaults;
+    }
+
+    Object.keys(lsdata).forEach((id) => {
+      state[`${id}Visible`] = lsdata[id].active;
+    });
+
+    this.setState(state, this.saveViews);
+  }
+
+  saveViews = () => {
+    const lsdata = this.getDefaultViews();
+
+    Object.keys(lsdata).forEach((id) => {
+      lsdata[id].active = this.state[`${id}Visible`];
+    });
+
+    window.localStorage.setItem(LS_VIEWS, JSON.stringify(lsdata));
```

Comment:
e.g. exception if sandboxed. Yes, not a bad idea I think we will probably start (ab)using it a bit more apart from the current tooltips, signer & views config. Should do it properly. (Going to log that seperately to split these out, since I wouldn't just want to do it for the one place and the rest don't fit in this PR)